backend/logic/shop/health.js

```diff
@@ -0,0 +1,332 @@
+    const { missingKeys, unknownKeys } = checkConfig(
+      method,
+      offlinePaymentBaseKeys,
+      offlinePaymentsOptionalKeys
+    )
+    if (missingKeys.length > 0) {
+      log.error(
+        `Shop {shop.id} - ${configName} misses offline payment keys ${missingKeys}`
+      )
+      errors.push(
+        `Method ${method.label}: DB config misses keys ${missingKeys}`
+      )
+    }
+    if (unknownKeys.length > 0) {
+      log.error(
+        `Shop {shop.id} - ${configName} includes unexpected keys ${unknownKeys}`
+      )
+      errors.push(`DB config includes unexpected keys ${unknownKeys}`)
+    }
+  }
+  return errors
+}
+
+/**
+ * Runs a battery of tests on a shop to identify any potential
+ * configuration or data related issues.
+ *
+ * @param {models.Shop} shop
+ * @returns {object}
+ */
+async function diagnoseShop(shop) {
+  const diagnostic = {}
+  const network = await Network.findOne({ where: { active: true } })
+  const networkConfig = getConfig(network.config)
+  const shopIsPublished = await ShopDeployment.findOne({
+    where: {
+      shopId: shop.id,
+      status: ShopDeploymentStatuses.Success
+    }
+  })
+
+  //
+  // Decrypt the encrypted config from the DB and check its validity.
+  //
+  let shopConfig = {}
+  let errors = []
+  try {
+    shopConfig = getConfig(shop.config)
+    const { missingKeys, unknownKeys } = checkConfig(
+      shopConfig,
+      dbConfigBaseKeys,
+      dbConfigOptionalKeys
+    )
+    if (missingKeys.length > 0) {
+      log.error(
+        `Shop {shop.id} - json config on disk misses some keys ${missingKeys}`
+      )
+      errors.push(`Json config on disk misses keys ${missingKeys}`)
+    }
+    if (unknownKeys.length > 0) {
+      log.error(
+        `Shop {shop.id} - json config on disk includes unexpected keys ${unknownKeys}`
+      )
+      errors.push(`Json config on disk includes unexpected keys ${unknownKeys}`)
+    }
+  } catch (e) {
+    log.error(`Shop {shop.id} - Failed loading shop config: ${e}`)
+    diagnostic.dbConfig = {
+      status: 'ERROR',
+      error: "Failed loading shop's config"
+    }
+  }
+
+  //
+  // Load config.json from disk and check its validity.
+  //
+  errors = []
+  let shopJsonConfigOnDisk = {}
+  try {
+    shopJsonConfigOnDisk = loadJsonConfigFromDisk(shop)
+    errors = checkJsonConfig(
+      shop.id,
+      network.networkId,
+      shopJsonConfigOnDisk,
+      'Json config on disk'
+    )
+  } catch (e) {
+    log.error(`Shop {shop.id} - Failed loading json config from disk: ${e}`)
+    errors.push(`Failed loading json config from disk`)
+  }
+  if (errors.length > 0) {
+    diagnostic.jsonConfigOnDisk = { status: 'ERROR', errors }
+  } else {
+    diagnostic.jsonConfigOnDisk = { status: 'OK' }
+  }
+
+  //
+  // If the shop is published, load config.json from the web and check its validity.
+  //
+  errors = []
+  let shopJsonConfigOnWeb = {}
+  if (shopIsPublished) {
+    try {
+      const url = `shopConfig.dataUrl/config.json`
```

Comment:
missing the wrapper(`${shopConfig.dataUrl}`) here?

Reply:
oops! good catch